libs/base/core.cpp

```diff
@@ -655,13 +655,13 @@ String substr(String s, int start, int length) {
 int indexOf(String s, String searchString, int start) {
     if (!s || !searchString)
         return -1;
-    auto lenA = s->getUTF8Size();
+    auto lenA = s->getUTF8Size() - start;
```

Comment:
start is the number of character, which might be greater than start in bytes.

Reply:
Ah, so this would work in my case as I was using characters that would fit into a single byte, but break for e.g. chinese characters that would be split across a few bytes?

Is there a helper for length of a single unicode character or something similar for this? It looks like the current behavior is just doing a `memcmp` a few extra times for the partial characters, which won't work when we actually need the character counts here?

Reply:
use `getUTF8DataAt(start)` - this gives pointer, you can subtract from that `getUTF8Data()` to get offset

`memcmp()` is fine, since no UTF8 character is a substring of another UTF8 character.